phpcs.xml.dist

```diff
@@ -3,9 +3,6 @@
 	<description>Sniffs for WordPress plugins, with minor modifications for Gutenberg</description>
 
 	<config name="testVersion" value="5.6-"/>
-	<rule ref="PHPCompatibilityWP">
-		<include-pattern>*\.php$</include-pattern>
-	</rule>
 
 	<rule ref="WordPress-Core"/>
 	<rule ref="WordPress-Docs"/>
```

Comment:
Core doesn't use the `WordPress-Docs` ruleset. For better parity, may want to consider removing this ruleset or at a minimum discussing why it's needed.

Reply:
The Docs ruleset doesn't examine code, only documentation.

Reply:
Hey @jrfnl,
Can you take a look at these comments, which are the source of why I first raised the question:

https://github.com/WordPress/gutenberg/pull/48603#issuecomment-1453254910
https://github.com/WordPress/gutenberg/pull/48603#issuecomment-1453511635

What do you think?

Reply:
In both cases mentioned [here](https://github.com/WordPress/gutenberg/pull/48603#issuecomment-1453254910), the docs in the GB repo are better and the docs in WP Core should be updated.

As Core isn't (yet) using the Docs ruleset, there is no reason why those updates should conflict with existing rules in WP Core.

Reply:
@jrfnl Yes, but `WordPress-Docs` causes linter errors when code gets backported from Core to Gutenberg, making it harder to keep both projects in sync (as [the comment](https://github.com/WordPress/gutenberg/pull/48603#issuecomment-1453254910) above showed).

Based on my understanding, WordPress-Docs could be temporarily disabled to make the backporting process simpler, and then re-enabled when Core supports it.

A decision needs to be made regarding whether the simplicity of backporting code takes precedence over enforcing the WordPress-Docs standard.

Reply:
>In both cases mentioned https://github.com/WordPress/gutenberg/pull/48603#issuecomment-1453254910, the docs in the GB repo are better and the docs in WP Core should be updated.

The docs may be better, but with this linting ruleset, they do not have parity with Core. 

But without the ruleset in Core, how will contributors know Core's changes comply before porting the code back to Gutenberg? They won't. Thus it will cause more work and potentially further cause the code to be out-of-sync.

The goal of this PR is to achieve Core parity. Once Core introduces this ruleset, then Gutenberg can once again add it back into its workflow. But until then, this ruleset means Gutenberg and Core do not have parity and thus, more work is required as noted when porting Core changes to Gutenberg.

IMO this ruleset needs to be removed from Gutenberg to achieve Core parity.

Reply:
Also noting, this ruleset needs to first be adopted in Core. Then it can be (re)adopted in Gutenberg.

Reply:
https://github.com/WordPress/gutenberg/issues/56487